Some Docs suggestions (#114)
thautwarm/MLStyle.jl: docs/what.md

## Motivations

> Those used to functional programming may feel limited when they don't have access to pattern matching and ADTs. Of course I'm one of those people.

Reviewer: I don't think this change helps much

Author: Yea true I'll roll this one back

In recent years I was addicted to extending Python with metaprogramming and even internal mechanisms. Although I made some interesting packages like [pattern-matching](https://github.com/Xython/pattern-matching), [goto](https://github.com/thautwarm/Redy/blob/master/Redy/Opt/builtin_features/_goto.py), [ADTs](https://github.com/thautwarm/Redy/tree/master/Redy/ADT), [constexpr](https://github.com/thautwarm/Redy/blob/master/Redy/Opt/builtin_features/_constexpr.py), [macros](https://github.com/thautwarm/Redy/blob/master/Redy/Opt/builtin_features/_macro.py), etc., most of these implementations are also disgustingly evil. Fortunately, in Julia, these features could be achieved straightforwardly without using any black magic. At last, some of these ideas come into existence with MLStyle.jl.

> Finally, we have such a library that provides **extensible pattern matching** inside of an efficient language.

Reviewer: ```suggestion
Finally, we have such a library that provides **extensible pattern matching** for julia.
```
